xbmc/guilib/GUIFontManager.cpp

```diff
@@ -158,20 +157,25 @@ CGUIFont* GUIFontManager::LoadTTF(const std::string& strFontName,
       // font could not be loaded - try Arial.ttf, which we distribute
       if (strFilename != "arial.ttf")
       {
-        CLog::Log(LOGERROR, "Couldn't load font name: {}({}), trying to substitute arial.ttf",
-                  strFontName, strFilename);
-        return LoadTTF(strFontName, "arial.ttf", textColor, shadowColor, iSize, iStyle, border, lineSpacing, originalAspect);
+        CLog::Log(
+            LOGERROR,
+            "GUIFontManager::{}: Couldn't load font name: {}({}), trying to substitute arial.ttf",
+            __func__, strFontName, strFilename);
+        return LoadTTF(strFontName, "arial.ttf", textColor, shadowColor, iSize, iStyle, border,
+                       lineSpacing, originalAspect);
       }
-      CLog::Log(LOGERROR, "Couldn't load font name:{} file:{}", strFontName, strPath);
+      CLog::Log(LOGERROR, "GUIFontManager::{}: Couldn't load font name:{} file:{}", __func__,
+                strFontName, strPath);
 
-      return NULL;
+      return nullptr;
     }
 
     m_vecFontFiles.push_back(pFontFile);
   }
 
   // font file is loaded, create our CGUIFont
-  CGUIFont *pNewFont = new CGUIFont(strFontName, iStyle, textColor, shadowColor, lineSpacing, (float)iSize, pFontFile);
+  CGUIFont* pNewFont = new CGUIFont(strFontName, iStyle, textColor, shadowColor, lineSpacing,
+                                    (float)iSize, pFontFile);
   m_vecFonts.push_back(pNewFont);
```

Comment:
Just a comment, but maybe an opportunity to change from a new/delete call into a vector of smart pointers. at a glance the new/delete stuff all line up, so probably not absolutely required.

Reply:
Have added on new commit an change to use `shared_ptr` about.

On tests it was OK, only found a memleak about `CGUIButtonControl` by change of language within settings.

One class is not removed and then remains in Kodi as a memleak. Could narrow it down in search to seems likely to come from CGUIDialogSettingsBase (presumably here https://github.com/xbmc/xbmc/blob/master/xbmc/settings/dialogs/GUIDialogSettingsBase.cpp#L371, but not sure?)

It would probably be good to revise the controls in another request in order to also have `shared_mem` there, with the huge number of GUIControls it would be really useful and safer.